src/main/java/net/mcreator/ui/component/entries/JSimpleListEntry.java

```diff
@@ -57,12 +60,45 @@ public JSimpleListEntry(JPanel parent, List<? extends JSimpleListEntry<T>> entry
 			entryRemovedByUserHandler();
 		});
 
-		add(PanelUtils.centerAndEastElement(line, PanelUtils.join(remove)));
+		moveUp.setToolTipText(L10N.t("simple_list_entry.move_up"));
+		moveUp.addActionListener(e -> {
+			int i = entryList.indexOf(this);
+			if (i > 0) {
+				Collections.swap(entryList, i - 1, i);
+				swapEntries(parent, i - 1, i);
+				parent.revalidate();
+				parent.repaint();
+			}
+		});
+
+		moveDown.setToolTipText(L10N.t("simple_list_entry.move_down"));
+		moveDown.addActionListener(e -> {
+			int i = entryList.indexOf(this);
+			if (i >= 0 && i < entryList.size() - 1) {
+				Collections.swap(entryList, i, i + 1);
+				swapEntries(parent, i, i + 1);
+				parent.revalidate();
+				parent.repaint();
+			}
+		});
+
+		add(PanelUtils.westAndCenterElement(PanelUtils.join(moveUp, moveDown),
+				PanelUtils.centerAndEastElement(line, PanelUtils.join(remove))));
 
 		parent.revalidate();
 		parent.repaint();
 	}
 
+	private static void swapEntries(JPanel parent, int thisIndex, int otherIndex) {
+		Component[] components = parent.getComponents();

```

Comment:
This only changes UI order. Actual data is stored in `List<? extends JSimpleListEntry<T>> entryList` and this data needs to be re-ordered too